docs/extend.rst

```diff
@@ -665,6 +665,141 @@ the same adapter as before:
+interface. This interface allows you to handle the `update` operation by your
+own. This interface provides the folllowing signatures:
+
+.. code-block:: python
+
+    class IUpdate(interface.Interface):
+        """Interface to handle update of objects
+        """
+
+        def is_update_allowed(self):
+            """Returns whether the update of the object is allowed
+            """
+
+        def is_update_delegated(self):
+            """Returns whether the update of this object has to be delegated to
+            this adapter
+            """
+
+        def update_object(self, **data):
+            """Updates the object
+            """
+
+
+Allow/disallow to update an object
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For instance, say you don't want to allow the update of objects from type
+`Todo` through the `senaite.jsonapi`:
+
+.. code-block:: python
+
+    from senaite.jsonapi.interfaces import IUpdate
+    from zope import interface
+
+
+    class TodoUpdateAdapter(object):
+        """Custom adapter for the update of objects from Todo type
+        """
+        interface.implements(IUpdate)
+
+        def __init__(self, context):
+            self.context = context
+
+        def is_update_allowed(self):
+            """Returns whether the update of the object is allowed
+            """
+            return False
+
+
+Register the adapter in your `configure.zcml` file for your special interface:
+
+.. code-block:: xml
+
+    <configure
+        xmlns="http://namespaces.zope.org/zope">
+
+        <!-- Adapter for custom update -->
+        <adapter
+          factory=".TodoUpdateAdapter"
+          provides="senaite.jsonapi.interfaces.IUpdate"
+          for="my.addon.interfaces.ITodo" />
+
+    </configure>
+
+
+.. note::
+    This adapter is initialized with `context`, the object to be updated.
+
+.. note::
+    We've used here a custom `Todo` type, but you can use this approach for any
+    type registered in the system, being it from `senaite.core` (e.g. `Client',
+    `SampleType`, etc.) or from any other add-on.
+
+
+Custom update of an object
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Imagine that besides updating your object, you want to add a `Remarks` at the
+same time. You can use the same adapter as before:
+
+.. code-block:: python
+
+    from senaite.jsonapi.interfaces import IUpdate
+    from zope import interface
+
+
+    class TodoUpdateAdapter(object):
+        """Custom adapter for the update of objects from Todo type
+        """
+        interface.implements(IUpdate)
+
+        def __init__(self, context):
+            self.context = context
+
+        def is_update_allowed(self):
+            """Returns whether the update of the object is allowed
+            """
+            return True
+
+        def is_update_delegated(self):
+            """Returns whether the update of this object has to be delegated to
+            this adapter
+            """
+            return True
```

Comment:
Isn't that always the case when I've written an adapter?
Probably it would make more sense to explicitly call the update method from the api module when I find out in this method that rather the original update method should be called ...

Reply:
Might not be always the case. Imagine that you only want `is_update_allowed` to return False/True depending on your "custom" logic, but if returns True, you are ok with senaite.jsonapi` handling the update process by its own. Another option would be you want the update process to be custom-handled in your add-on only when certain conditions are met. Understand, but not sure, I would leave it as it is for now.